tests/integration-tests/tests/api/test_api.py

```diff
@@ -99,10 +101,18 @@ def _cloudformation_wait(region, stack_name, status):
     waiter.wait(StackName=stack_name, WaiterConfig={"MaxAttempts": 180})
 
 
+def _ec2_wait_running(region, instances):
+    _ec2_wait(region, instances, "running")
+
+
 def _ec2_wait_terminated(region, instances):
+    _ec2_wait(region, instances, "terminated")
+
+
+def _ec2_wait(region, instances, status):
     config = Config(region_name=region)
     ec2 = boto3.client("ec2", config=config)
-    waiter = ec2.get_waiter("instance_terminated")
+    waiter = ec2.get_waiter(f"instance_{status}")
```

Comment:
minor: since this is a generic `get_waiter` and not just a getter for `instance_*` waiters, I would keep this generic, and pass the full waiter name from the outside

Reply:
Agreed.